### ISSUE TITLE
The tests - particularly the drugs/ZINC compounds - are too slow

### ISSUE THREAD

**Issue by davidlmobley** [choderalab/openmoltools#109]
@kyleabeauchamp - I commented earlier by e-mail that the tests seem to be running far too slow for me. Particularly, the "Testing drug ZINC#####..." tests have now been running for several days on my current (2015 model) 13" MBP. Progress is being made but there seem to be far too many compounds. Several issues come to mind:

* Is there a good reason to test so many ZINC compounds? Can we pare this number down so the tests will run in a reasonable amount of time?
* From what I can tell, it should be possible to configure the tests so slow tests can be skipped; can we do this? (Currently I don't seem to be able to skip them, though they are flagged as slow).
* You mentioned earlier that the tests are "slow" for you also, but is "several days" within the expected range or is there something wrong on my end?

**Comment by kyleabeauchamp** [comment]
So I'm not sure these slow tests are particularly useful to begin with, as they're testing a branch (OpenMM ffXML) that is in an experimental and untested state to begin with.  

I personally would rather see tests of the AMBER / Gromacs / ACPYPE / OpenEye functions, as those are the things that most end users (and we) will be using.

**Comment by davidlmobley** [comment]
I'm not experienced with using this type of tests yet, but is there a way we can set a flag so these won't run by default and will only be run if someone wants them? I don't really want to have to wait several days to find out if my new code passes the tests each time because some old tests for code we don't use are taking a really long time... :)

**Comment by rmcgibbo** [comment]
Yeah, here's an example:

```
$ cat test.py
from numpy.testing.decorators import slow

@slow
def test_1():
    print('running test_1')

def test_2():
    print('running test_2')
```

**Comment by rmcgibbo** [comment]
and then, running `nosetests` will run both tests,
```
$ nosetests test.py -s
running test_1
.running test_2
.
----------------------------------------------------------------------
Ran 2 tests in 0.001s

OK
```

but running all 'non-slow' tests will only run the second:
```
$ nosetests test.py -s -a '!slow'
running test_2
.
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK
```

**Comment by rmcgibbo** [comment]
Some docs: http://nose.readthedocs.org/en/latest/plugins/attrib.html

**Comment by davidlmobley** [comment]
@rmcgibbo - I actually saw that. AND the tests in question are already flagged as slow as far as I can tell, but 

`nosetests openmoltools -v -exe -s -a '!slow'`

still seems to execute them.

I'll look at the code some more and see if I can tell why.

**Comment by davidlmobley** [comment]
One of the slow functions was missing the `slow` attribute for some reason. I've issued a PR to fix it, so at least now this can be skipped. 

The larger issue still remains though - the default behavior should be to skip these particular slow tests since they are not useful. (Specifically, if a rookie user - like me - follows the prescribed command from the README.md file displayed on the main page, the tests will currently take several days to run with most of that spent testing code they won't need tested.) And I don't yet know how to change the default behavior.

**Comment by rmcgibbo** [comment]
Yeah, that would be better. This solution is not great, but might help: http://stackoverflow.com/questions/14674713/nose-how-to-skip-tests-by-default

**Comment by davidlmobley** [comment]
OK. What we probably should do is flag these with a second attribute, something like `experimental` or similar, and we can have the default behavior to skip the `experimental` tests. Some slow tests will be needed probably, but these are problematic because they are not only slow but also not important. @kyleabeauchamp , how does that sound?

**Comment by kyleabeauchamp** [comment]
Yeah, that sounds reasonable.

**Comment by kyleabeauchamp** [comment]
I think your pull request solved this issue, feel free to reopen if I am incorrect.

**Comment by davidlmobley** [comment]
No, my PR just addressed the issue where one of the tests which was slow was not flagged as slow. We still have to:
* Flag the unimportant slow tests as `experimental`
* Apply @rmcgibbo 's solution so by default the `experimental` tests are skipped

Thanks.
